examples/web_sys/reactive-state/src/subscriber.rs

```diff
@@ -0,0 +1,79 @@
+use futures::future::ready;
+use futures_signals::signal::SignalExt;
+use wasm_bindgen_futures::spawn_local;
+use yew::prelude::*;
+
+use super::store::{ArcState, Store, StoreOutput};
+
+pub struct Subscriber {
+    ip: Option<String>,
+    link: ComponentLink<Subscriber>,
+    store: Box<dyn Bridge<Store>>,
+    state_ref: Option<ArcState>,
+    id: i32,
+}
+
+pub enum Msg {
+    FromStore(StoreOutput),
+    SetIp(Option<String>),
+}
+
+impl Subscriber {
+    fn register_state_handlers(&self) {
+        let callback = self.link.callback(|ip| Msg::SetIp(ip));
+        let state = self.state_ref.as_ref().unwrap();
+        let handler = state.ip.signal_cloned().for_each(move |u| {
+            callback.emit(u);
+            ready(())
+        });
+        spawn_local(handler);
+    }
+}
+
+#[derive(Properties, Clone)]
+pub struct Props {
+    pub id: i32,
+}
+
+impl Component for Subscriber {
+    type Properties = Props;
+    type Message = Msg;
+
+    fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+        let store = Store::bridge(link.callback(|d| Msg::FromStore(d)));
+        Self {
+            ip: None,
+            link,
+            store,
+            state_ref: None,
+            id: props.id,
+        }
+    }
+
+    fn update(&mut self, msg: Self::Message) -> ShouldRender {
+        match msg {
+            Msg::FromStore(s) => match s {
+                StoreOutput::StateInstance(state) => {
+                    self.state_ref = Some(state);
+                    self.register_state_handlers();
+                }
+            },
+            Msg::SetIp(ip) => self.ip = ip,
+        }
+        true
+    }
+
+    fn view(&self) -> Html {
+        let ip = if self.ip.is_some() {
+            self.ip.as_ref().unwrap()
+        } else {
+            "No IP yet. It will show up here"
+        };
```

Comment:
So doing that adds a bit of an issue with mismatched types :( the unwrap value in this case is a `String` but the string literal is of course `&'static str`. the `unwrap_or` is expecting a `&String` so not even `.to_string` is helping.

I got around this by assigning it to a variable first: 
```
        let ip_message = "No IP yet. It will show up here".to_string();
        let ip = self.ip.as_ref().unwrap_or(&ip_message);
```

Not much cleaner this way, but it compiles and runs

Reply:
Ah sorry about that! This works:
```rust
        let ip = self.ip.as_deref().unwrap_or("No IP yet. It will show up here");
```
`String` can be dereferenced to `&str` :)